docs/eBpfStringsAPI.md

```diff
@@ -0,0 +1,117 @@
+through the Accept-Encoding header, because some older application can't handle gzip appropriately. (I'm sure this is a
+real use case somewhere.) Error handling is elided in this example for brevity.
+
+```C
+char accept_encoding[] = "Accept-Encoding";
+int ae_len = sizeof(accept_encoding);
+char gzip_name[] = "gzip";
+int gzip_len = sizeof(gzip_name) - 1; // exclude terminal null
+char *header_body = NULL;
+int header_length = ebpfHttpLoadHeader(ctx, accept_encoding, ae_len, &header_body);
+
+char *gzip_location = ebpf_strstr(header_body, header_length, gzip_name, gzip_len);
+// if gzip isn't first, copy everything before it back to the header, then copy everything after it back to the header.
+if (gzip_location > header_body)
+{
+    ebpfHttpSetHeader(ctx, accept_encoding, ae_len, header_body, (gzip_location - header_body));
+}
+
+char *after_header = gzip_location + gzip_len;
+int after_size = header_length - ((gzip_location - header_body) + gzip_len);
+while (after_size > 1)
+{
+    // This loop has to check for at least a terminal null, so it requires a length > 1
+    if (*after_header == ',' || *after_header == ' ')
+    {
+        after_size--;
+        after_header++;
+        continue;
+    }
+    else
+    {
+        break;
+    }
+}
+
+if (after_size > 1)
+{
+    ebpfHttpAppendHeader(ctx, accept_encoding, ae_len, after_header, after_size);
+}
+```
+
+## Functions of interest
+
+We should aim to have sufficient coverage of the core C string functions. The functions that are of interest in
+particular are listed here; their eBPF counterparts will necessarily need to be somewhat different, especially an
+example like `strstr()`.
+
+```C
+errno_t strncpy_s(char *restrict dest, size_t dest_size, const char *restrict src, size_t count);
+errno_t strncat_s(char *restrict dest, size_t dest_size, const char *restrict src, size_t count);
+size_t strnlen_s(const char *str, size_t str_size);
+int strncmp(const char *lhs, const char *rhs, size_t count);
+char *strchr(const char *str, int ch);
+char *strstr(const char *str, const char *substr);
+```
+
+Barring a strong requirement, this proposal is limited to 8-bit and UTF-8 strings. Wide strings would follow a similar
+pattern, but are not considered necessary here.
+
+While there is an upstream `bpf_strncmp()`, its argument pattern is inconsistent with the C stdlib `strncmp()`, and the
+lack of an argument for the second string's length makes it not fit the pattern of our eBPF extension functions. That
+appears to be implemented as another type of argument that our runtime doesn't allow for currently. As such, our
+implementation will not be compatible with upstream's `bpf_strncmp()` but will instead use `bpf_strcmp()` so as to not
+cause an uncomfortable name collision.
+
+## Proposed names & prototypes
+
+```C
+errno_t bpf_strcpy(char *restrict dest, size_t dest_size, const char *restrict src, size_t count);
```

Comment:
For functions that don't already exist on linux we should use the _s suffix like bpf_strcpy_s.
That's safer in case Linux adds bpf_strcpy with a different signature.

Reply:
I've matched function names more closely and added _s suffixes.